cloudbridge/providers/gcp/provider.py

```diff
@@ -230,6 +230,8 @@ def __init__(self, config):
 
         if self.credentials_dict and 'project_id' in self.credentials_dict:
             self.project_name = self.credentials_dict['project_id']
+        elif self._get_config_value("project_id"):
```

Comment:
It seems that project_id is already being read in from the credentials_dict, so it looks like we can just reuse that.
```
 config = {
    'gcp_credentials_obj': credentials,
    'gcp_credentials_dict': { 'project_id': '<project_id>'}
  }
```
This will work because credentials_obj always takes precedence over credentials_dict, so passing in both shouldn't cause problems.

Reply:
Correct; though a flat dictionary may read better than a nested one.